src/Requester.cpp

```diff
@@ -0,0 +1,73 @@
+#include "Requester.h"
+#include <iostream>
+
+static void worker(zsock_t *pipe, void *args);
+
+Requester::Requester() {
+    // start a new backround thread 
+    // to process incoming/outgoing requests
+    worker_ = zactor_new(worker, NULL);
+}
+
+Requester::~Requester() {
+    zactor_destroy(&worker_);
+}
+
+zmsg_t* Requester::request(zmsg_t **msg) {
+    zmsg_pushstr(*msg, "REQUEST");
+    zmsg_send(msg, worker_);
+    return zmsg_recv(worker_);
+}
+
+std::stringstream Requester::send(std::string task_name, std::string data) {
+    zmsg_t *msg = zmsg_new();
+    zmsg_addstr(msg, "TASK");
+    zmsg_addstr(msg, task_name.c_str());
+    zmsg_addstr(msg, data.c_str());
+
+    zmsg_t *reply = request(&msg);
```

Comment:
is the allocated zmsg object adopted by this function? who will cal zmsg_destroy() for it?

Reply:
If you are talking about the `msg` variable (object) then it will be destroyed after being sent in `request()` method (by `zmsg_send` function according to czmq docs). The possible issue I see now is that `request()` method hides that it will destroy `msg`. Maybe, adding a comment that `method will destroy message after it sent` will raise awareness.

Reply:
yes but if you had a c++ wrapper for this, the could would look like

ZMsg msg;
msg.add_str(task_name);
msg.add_str(data);
request(std::move(msg));

and then there would be no question, as the move conveys this information (even if the ZMsg is non-copiable).